### ISSUE TITLE
Is there a way to force large (graphical) episode names on the menu?

### ISSUE THREAD

**Issue by maverick-hunter** [Owlet7/wadfusion#6]
Guys, I don't even know if this project was already supposed to be used by anyone, but I've been following the commits after you picked up after WadSmoosh ended development. Thank you so much for that! 

I was just wondering if, with so many more episodes now, there's still a way to keep the classic Doom font for the the episode list, instead of the plain text used by GzDoom.

I don't really know if it fits the goals of the project, but I was wondering if you'd consider an option for submenus for Doom 1 episodes and Doom 2, sincer there are so many official episodes now.

Also, I think it would be cool if the command prompt gave a message about which wads one's missing for the complete package.

Anyway, thanks again for keeping this project alive!

**Comment by Trimint123** [comment]
Considering that it's pretty much impossible to have every custom text in one screen, I think it's safe to assume that the idea for it wasn't planned unless we find a way to make the menu scroll, which at that point we may as well make a fancier menu instead.

**Comment by maverick-hunter** [comment]
I don't even know if it's doable, but would submenus somewhat like this be out of the scope of the project? Maybe

+ Doom
    - Knee-Deep In The Dead
    - The Shores of Hell
    - Inferno
    - Thy Flesh Consumed
    - Sigil
    - Sigil II

+ Doom II
    - Hell on Earth
    - No Rest For The Living
    - The Master Levels
    - No Rest for the Living
    - The Vulcan Abyss
    - Counterfeit Eden
     
+ Final Doom
    - TNT Evilution
    - The Plutonia Experiment

**Comment by Trimint123** [comment]
If possible, maybe we could split sections of Hell on Earth, TNT, and Plutonia to be episodic, though a creative name is required for those since they ain't an official names.

**Comment by Owlet7** [comment]
That's the plan, more or less. The episodes should be grouped by game and campaign.
The Hell on Earth episodes do have official names: "The Space Station", "The City", and "Hell".

> Also, I think it would be cool if the command prompt gave a message about which wads one's missing for the complete package.

That would lead to some potential redundancies (e.g. saying that doom2kex is missing when the user already uses it as doom2). It would also clutter the screen with too much information. The "Supported WADs" section of the Readme should suffice, it already covers everything that's supported.

![menus_doom_fusion](https://github.com/user-attachments/assets/44a3a455-87db-4320-b925-ae108518bda5)

**Comment by Trimint123** [comment]
Ohh, you planned support for [Works of the Masters](https://jp.itch.io/deluxe-master-levels). Interesting!

As for the episodes name, I don't know. Some of the names is a bit uncreative, but I guess we can't do much about it, huh?

**Comment by Trimint123** [comment]
I do have some name ideas in mind for Hell of Earth:
- The Space Station
- Chaos in the City
- Return to Hell

**Comment by Owlet7** [comment]
> Ohh, taken ideas from [Works of the Masters](https://jp.itch.io/deluxe-master-levels). Interesting!

Actually, no. I made [a similar mod](https://steamcommunity.com/sharedfiles/filedetails/?id=237853755) 10 years ago, long before WadSmoosh, and before ZScript was implemented in GZDoom. I thought that it'd fit the goals of the project to add support for these maps.

> As for the episodes name, I don't know. Some of the names is a bit uncreative, but I guess we can't do much about it, huh?

I wouldn't want to change the Doom II episode names, since those are official, but Final Doom doesn't have official names for its episodes, as far as I'm aware, so I'm open to suggestions.

**Comment by maverick-hunter** [comment]
> That's the plan, more or less. The episodes should be grouped by game and campaign. The Hell on Earth episodes do have official names: "The Space Station", "The City", and "Hell".

That would look really cool, IMHO. Thanks for sharing that graph!

**Comment by Trimint123** [comment]
> I wouldn't want to change the Doom II episode names, since those are official, but Final Doom doesn't have official names for its episodes, as far as I'm aware, so I'm open to suggestions.

TNT is about alien demons invaded earth through a spaceship, so it should be space related. The Doom Wiki had some interesting idea of how TNT's episodic should structure:
MAP1-11: experiment lab, then millitary base
MAP12-20: nighttime (may as well be in space considering the next one)
MAP21-30: hellish (the demon's planet, probably)

For Plutonia, I'm not sure how I would theme that one specifically.

**Comment by Owlet7** [comment]
TNT is set on Io, a moon of Jupiter, not on Earth.

**Comment by Trimint123** [comment]
Huh

**Comment by RoniPunk** [comment]
Amazing work on WAD Fusion! I too was working on something to fix the game index. I am also working on a full run mod for Wad Fusion as there was one for wadsmoosh and also I added the PSX doom exclusive maps as hidden extras as the Xbox version did for Sewers and Betray.

**Comment by Snarke1** [comment]
Oooh PSX levels would be an interesting addition. Though, do those levels even have any way to get them without plucking them out of a rom or something?

**Comment by maverick-hunter** [comment]
> Oooh PSX levels would be an interesting addition. Though, do those levels even have any way to get them without plucking them out of a rom or something?

Lookup Doom CE on ModDB. I know you need the official release for the Doom 64 stuff, but I'm not sure the copyright stuff related to PSX Doom. 

Even so, I think both Doom 64 and PSX Doom are too different from the originals to fit within WadFusion's scope, maybe?

**Comment by Trimint123** [comment]
Yeah, I don't think it was necessary to include here. If you know how to code tho, you could do it for yourself.

**Comment by Snarke1** [comment]
I meant more the exclusive secret levels specifically but yeah I getcha. Also I wish I knew how to code and add stuff to this lmao! It's fine though, this project is still very appreciated.

**Comment by Owlet7** [comment]
Just to make it clear, none of the 5th gen console Doom content will ever be officially supported in WadFusion, either in full, or only "exclusive" levels.

**Comment by Snarke1** [comment]
Yeah that's fair. My bad. I got a little too excited about the idea. Regardless, I look forward to whatever comes next!

**Comment by Ohthatguy43** [comment]
will the order be smth like this?

DOOM: (Ultimate DOOM + Sigil 1/Sigil 2)

DOOM 2: (DOOM 2, NO REST FOR THE LIVING AND MASTER LEVELS)

FINAL DOOM: (TNT AND PLUTONIA)

LEGACY OF RUST: (The Two Episodes featured.)

**Comment by Someguyman** [comment]
> Planned menu structure
> ![](https://private-user-images.githubusercontent.com/83693185/367532555-e1a461d8-29cf-49cd-abd1-e2f5ca379724.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk3MDAzNTUsIm5iZiI6MTcyOTcwMDA1NSwicGF0aCI6Ii84MzY5MzE4NS8zNjc1MzI1NTUtZTFhNDYxZDgtMjljZi00OWNkLWFiZDEtZTJmNWNhMzc5NzI0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIzVDE2MTQxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdhZjg2MWVhMmQ2ZmNhNmUwNjFjZDAzYzgyZmNmYWRiNjFjOGExZTlkNmQ1MGFiODFmYjc5Nzg4OWZlZDM5NTImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.g2Ah086YoxJbwA7riGpMPy-r_etH5a_etV2LlhIDiSI)

I suggestions for the tnt and plutonia episdoes:

TNT saga: 
Episode 1: Lo invasion
Episode 2: Nocturnal Lo
>Episode 3 remains as "Hell on lo". 

Plutonia saga:
Episode 1: Jungle Assault. I named it that due to the general vibe of plutonia with it's overall brown and green color palette and use of vines/camo patterned textures. 
Episode 2: The Blood Tides. Named after the sky texture of these maps and also to differentiate it from doom 2's hell episode. 
>Episode 3 remains as "Devil Hive".

**Comment by Someguyman** [comment]
I think it could also be worth it to use the new Label feature that was recently added to gzdoom's mapinfo lump so that you change the label from tn_map01 to something like Evilution, map01 without changing the filename itself.

**Comment by Owlet7** [comment]
Won't be done, because it overrides the "Show map label" automap option.

**Comment by Owlet7** [comment]
I'm brewing up an absolutely *disgusting* hack to make this whole thing possible right now. All the "new game" menus in GZDoom are hardcoded, so I'm gonna have to replace them with scripting that doesn't use MAPINFO definitions. I'll keep the original menu accessible from the mod menu.

**Comment by maverick-hunter** [comment]
That sounds really cool. I'm a complete ignorant regarding this, but what difference would the MAPINFO make for this?

**Comment by Owlet7** [comment]
The new menu wouldn't be able to account for any new episodes and skills that PWADs can add or replace.

**Comment by Owlet7** [comment]
I'm very demoralised by how hardcoded the "New Game" menus are in GZDoom, and how any attempts I come up with to hack around them result in implementations that are quite messy and ugly.

Through experimentation I have arrived at a solution that's somewhat tolerable, but I would like to get some feedback before I continue working on it.

[wadfusion_newgame_menu_test1.zip](https://github.com/user-attachments/files/18820466/wadfusion_newgame_menu_test1.zip)

You will need the latest GZDoom dev build and a (preferably complete) ipk3 generated from the latest WadFusion dev build. I only bothered implementing menus for the Doom 1 episodes for this test.

Some of the visible compromises:
- No custom episode and skill definitions will appear in these menus.
- PatchItem elements can't be localised, so I used invisible TextItem elements on top of StaticPatches. For some reason, these StaticPatchs don't display the localisation strings anyway. Is it a bug with GZDoom?
- The skill menu doesn't select "Hurt me plenty" by default.
- After selecting a map and skill, the game will first load an empty, completely black map before instantly transitioning to the selected map.

Are the compromises I arrived at acceptable? I may be able to do more to better hide the black screen, I'll probably post another test later.

**Comment by maverick-hunter** [comment]
The menu does look good and the black level transition is so fast that I don't think anyone would mind it. Thinking about it as a way to add all the official and official-adjacent content keeping the original aesthetics, I think it's pretty spot on. Would it be possible to make the level select optional?

Out of curiosity: to make this idea more flexible without hacking it in, some support would need to be coded into gzdoom?

**Comment by Owlet7** [comment]
> Would it be possible to make the level select optional?

I'd really like to make it optional, but that's another one of the compromises. That part of the menus is used to set which map / episode to load by setting a console command variable. Commands can only be sent from OptionMenu type menus (the ones with smaller font text), not from ListMenu type menus (all the other menus with big font text).

> Out of curiosity: to make this idea more flexible without hacking it in, some support would need to be coded into gzdoom?

Yes, but it would require making changes to the menu code and expanding ZMAPINFO. I wouldn't even know where to start, or if these would be desired changes.

**Comment by maverick-hunter** [comment]
> I'd really like to make it optional, but that's another one of the compromises. That part of the menus is used to set which map / episode to load by setting a console command variable. Commands can only be sent from OptionMenu type menus (the ones with smaller font text), not from ListMenu type menus (all the other menus with big font text).

As a compromise it's reasonable enough. I think it accomplishes the main idea just right.

> Yes, it would require making changes to the menu code and expanding ZMAPINFO. I wouldn't even know where to start, or if these would be desired changes.
 
I'm not really that into the doom mods scene, but I can't help but wonder if this idea of using submenus wouldn't be popular enough considering how many excellent doom mods and total conversions there are that could use it. Also WadSmoosh (and by extent Fusion now) are quite popular and as far as I know, GZDoom is more about quality of life than being faithful like something like chocolate doom. Maybe something could come out by bringing this up with GZDoom devs? Would that be too farfetched?

I mean, for the intents of grouping up the episodes and making it look good, it does look almost perfect, but the lack of flexibility seems to be frustrating since you're going through some much effort to make Fusion have so many QoL options for all the official content.

Anyway, for me it's almost perfect as it is now. Thanks so much"

**Comment by Owlet7** [comment]
Just moving stuff from the TODO.md file to the relevant issues.

> ## Feature creep territory
> - Level select menu
>   - Hide secret levels until they are discovered
>   - Progression tracker
>   - Per-skill stats tracker
>   - Pistol start tracker
>   - Fast monsters tracker
>   - Use the game's `M_DOOM` graphic in the level select menu?
>     - DOOM
>     - DOOM II
>     - TNT: Evilution
>     - The Plutonia Experiment
>     - Legacy of Rust
>     - Master Levels (KEX version, duplicate DOOM II's otherwise)
>     - No Rest for the Living (KEX version, duplicate DOOM II's otherwise)
>     - SIGIL (KEX version, duplicate DOOM's otherwise)
>     - SIGIL II (make one)
>     - The Ultimate DOOM (make one)
> - Rename difficulty levels for Master Levels - Cabal episode
>   - Hey, not too rough -> Hurt me plenty
>   - Hurt me plenty -> Ultra-violence
>   - Ultra-violence -> Total carnage (using gold font)
> - Add unique TitleMusic

**Comment by Owlet7** [comment]
The custom New Game menu now uses the original New Game menu as part of the workaround, so custom skills and classes (and episodes, in a more janky way) should be supported. The black screen when starting a new game has been reduced to one frame of the titlescreen graphic, before it melts into the selected map. Hopefully I can have the second test build ready this weekend.

**Comment by maverick-hunter** [comment]
> >   * Use the game's `M_DOOM` graphic in the level select menu?
> >     
> >     * DOOM
> >     * DOOM II
> >     * TNT: Evilution
> >     * The Plutonia Experiment
> >     * Legacy of Rust
> >     * Master Levels (KEX version, duplicate DOOM II's otherwise)
> >     * No Rest for the Living (KEX version, duplicate DOOM II's otherwise)
> >     * SIGIL (KEX version, duplicate DOOM's otherwise)
> >     * SIGIL II (make one)
> >     * The Ultimate DOOM (make one)

This would look really cool. I feel like the whole idea of having sub-menus for each of the official chapters along with the original font is to make an actual doom I + II definitive edition that I don't feel like the nightdive's release really accomplishes. I think all the these "feature creep" ideas you listed go along well with this concept. I wonder if would be possible to also change the titlepic with that. Anyway, it's looking really promising so far. Great job!

**Comment by Owlet7** [comment]
I still have more stuff to add, but I'd like to share a second test with the stuff I changed so far, while I continue working on it.

[wadfusion_newgame_menu_test2.zip](https://github.com/user-attachments/files/19133745/wadfusion_newgame_menu_test2.zip)

The cvar `ml_intros` controls whether the intros (including story intermissions) are played. The test build still assumes you have a complete game.

**Comment by maverick-hunter** [comment]
> I still have more stuff to add, but I'd like to share a second test with the stuff I changed so far, while I continue working on it.
> 
> [wadfusion_newgame_menu_test2.zip](https://github.com/user-attachments/files/19133745/wadfusion_newgame_menu_test2.zip)
> 
> The cvar `ml_intros` controls whether the intros (including story intermissions) are played. The test build still assumes you have a complete game.

I'm getting the following error if I run wadfusion with the sigil osts (don't know which one is causing it, but I removed both and it worked.

![Image](https://github.com/user-attachments/assets/ca46d8a6-a729-4a3f-af1f-a78f972f199e)

I like the titlepic for each chapter and the placeholder for the story from the manual, looks pretty official. The one frame transition used to show a black screen instead of the title pic in the first test? 

I like it a lot so far. :)

**Comment by Owlet7** [comment]
> I'm getting the following error if I run wadfusion with the sigil osts

Read the crash log carefully again, it happened because you tried loading `doom_fusion.ipk3` twice (perhaps by drag and dropping it into the exe?).

> The one frame transition used to show a black screen instead of the title pic in the first test?

Yup, a black screen with a long wipe transition.

> I like the titlepic for each chapter and the placeholder for the story from the manual, looks pretty official. I like it a lot so far. :)

I'm glad you like it so far! There's still a lot to improve though. For one, making it compatible with any WAD configuration.

**Comment by maverick-hunter** [comment]
> Read the crash log carefully again, it happened because you tried loading `doom_fusion.ipk3` twice (perhaps by drag and dropping it into the exe?).
That must be it! My mistake! Thanks!

> 
> > The one frame transition used to show a black screen instead of the title pic in the first test?
> 
> Yup, a black screen with a long wipe transition.
> 
Just an idea, but what if instead of dissolving to a simple black screen or back to the title pic, it used something like a black screen with a small text like "press to continue" so this screen has some purpose instead of just being a workaround?

**Comment by Owlet7** [comment]
What would be the point of that?

**Comment by maverick-hunter** [comment]
> What would be the point of that?

Nevermind, right now, it's: New Game -> Which Game? -> Which Campaign? -> Which Episode -> Which Level? -> Difficulty -> **TitlePic** -> Text from the manual -> **Title Pic Transition**

I confused which titlepic was waiting for an input. If it was the last one a "press to start"prompt would make some sense. I think just a black screen in this second title pic transition (which is the workaround, right?) maybe would look cleaner than falling back to the title pic that was already shown before the texts from the manual screen? I don't really understand anything about how you're making this work, so I don't even know if it'd be worth the trouble since it's so minor and it looks very clean already. ^^"

**Comment by Owlet7** [comment]
Would it be adequate if the custom New Game menu was provided as an optional PWAD? I really can't think of an elegant way of implementing it without breaking every single mod that modifies the main menu.

**Comment by Someguyman** [comment]
> Would it be adequate if the custom New Game menu was provided as an optional PWAD?

I assume this also goes for the episode intros too, right? If so, I'm fine with that being a separate pwad if it's mod compatibility you're concerned with.

**Comment by Owlet7** [comment]
I can add the intros without braking stuff (as far as I can foresee). It's the method which I have to use to add the custom New Game menu which will break them (modifying the main menu, removing all the episode definitions).

**Comment by Owlet7** [comment]
I have to clear all the episodes in order for the custom menu to auto-select the only episode available -- the hack level that warps you to the levels selected from the custom menu.

If you'd try starting a new game from a mod that modifies the main menu, it'd instantly send you to the cast ending intermission.

**Comment by maverick-hunter** [comment]
Are there any downsides to adding it as an optional PWAD?

**Comment by maverick-hunter** [comment]
It seems like a reasonable compromise to me.

**Comment by maverick-hunter** [comment]
It looks pretty much perfect IMO. The only nitpick I could see coming up is, since you gotta select a level within the episode, the names of the levels for doom II and the master levels are their filenames. I can see people complaining about the secret and alternate levels also appearing on the main list, but it doesn't bother me personally, and I think you said this map list is something  there's no way around, right?

**Comment by Owlet7** [comment]
> The only nitpick I could see coming up is that, since you gotta select a level within the episode, the names of the levels for doom II and the master levels are their filenames instead of proper titles like the other episodes.

Oh, that's just unfinished, it's tedious stuff that I'd rather do last.

> I can see people complaining about the secret and alternate levels also appearing on the main list, but it doesn't bother me personally, and I think you said this map list is something there's no way around, right?

There probably are ways, but I'm not experienced enough to know them. Hopefully this entire feature can be redone at a later point with a much less jank solution.

**Comment by maverick-hunter** [comment]
I love it. It looks really professional. 🥇

**Comment by Owlet7** [comment]
Yeah, it's very outdated now.

There have been some recent changes to GZDoom that should help make this menu a lot more polished, so consider all the previous test versions to be failed attempts.

**Comment by Con-Rez** [comment]
> As for the episodes name, I don't know. Some of the names is a bit uncreative, but I guess we can't do much about it, huh?

Okay, I actually had some name ideas for these when I was seperating out play sessions of WadSmoosh with a friend through QZandronum. I decided upon them based on the end text from each chapter and related media released for the games at the time.

- DOOM II = 
   - Ep1: (Subverted Starport) - Themed industrial area
      - Name is inspired by the [text screen before MAP07](https://doomwiki.org/wiki/File:Text_screen_before_M07.png) 
         > "...and the Starport's technology is being subverted by their presence."
      - Alternative name can be Hostage Starport as well
         > "You can penetrate into the haunted heart of the starbase and find the controlling switch which holds earth's population hostage."
   - Ep2: (Heart of the City) - Themed destroyed city
      - Name is inspired by the [the final text of the  screen before MAP12](https://doomwiki.org/wiki/File:Text_screen_before_M12.png)
         > "The alien base is in the heart of your own home city."
      - Referred to in the next [text after finishing the chapter before MAP 21](https://doomwiki.org/wiki/File:Text_screen_before_M21.png)
         > "You are at the corrupt heart of the city,..."
   - Ep3: (Entryway Through Hell) - Themed Hell
      -  Name is inspired by the final [text of the screen before MAP 21](https://doomwiki.org/wiki/File:Text_screen_before_M21.png)
         > " You see no way to destroy the creatures' entryway on this side,..."
         > "...What do you care if you're got to go through hell to get to it?"

- FINAL DOOM = 
   - TNT: Evilution - 
      - Ep1: (A Diseased Ordnance) - Themed Experimental Labs and then Military Complex
         - Inspired from the [text screen before MAP07](https://doomwiki.org/wiki/File:Text_screen_before_M07_FD_TNTE.png)
            > "Ahead lies the military complex, now swarming with diseased horrors hot to get their teeth into you. With luck, the complex still has some warlike ordnance laying around."
      - Ep2: (Reaming a Blood Feast) - Themed Spaceship in Space (or Nighttime depending on who you ask)
         - Name is inspired by the [text screen before MAP12](https://doomwiki.org/wiki/File:Text_screen_before_M12_FD_TNTE.png)
            > "They might be planning a blood feast, but you're ready to ream out a whole herd if you have to."
         - An alternative name could be inspired by the title of the doom novel "[Infernal Sky](https://doomwiki.org/wiki/Doom_novels)" that served as a sequel to Hell on Earth. The novel is about traveling through space in doom back to the demon's home world to bring the fight to them, which is oddly related to what's taking place in this episode.
      - Ep3: (The Agonizing Echo of Hell) - Themed Alien Planet with Hellish Industrial Bases
         -  Inspired from the [text screen after MAP30](https://doomwiki.org/wiki/File:Text_screen_after_M30_FD_TNTE.png)
         - Alternative name "Take Some Names" inspired by the [final text on the screen before MAP21](https://doomwiki.org/wiki/File:Text_screen_before_M21_FD_TNTE.png)
            > "Hefting your gun, an evil grin trickles onto your face. Time to take some names."
   - The Plutonia Experiment
      - Ep1: (Keep Living)
         - Inspired from the [final text on the screen before MAP07](https://doomwiki.org/wiki/File:Text_screen_before_M07_FD_TPE.png)
            > "Keep moving, keep fighting, keep killing. Oh yes, keep living too."
      - Ep2: (Good at That Kind of Thing)
         - Inspired from the [final text on the screen before MAP12](https://doomwiki.org/wiki/File:Text_screen_before_M12_FD_TPE.png)
            > "You're good at that kind of thing." 
      - Ep3: (A Little Hell)
         - Inspired from the [final text on the screen before MAP21](https://doomwiki.org/wiki/File:Text_screen_before_M21_FD_TPE.png)
            > "Grinning evilly, you check your gear, and get ready to give the bastard a little hell of your own making!"

By no means official names, but at least they have their reasons and fit with the tone of the episodes.

**Comment by maverick-hunter** [comment]
Just caught up with the whole GZDoom/UZDoom debacle and was just wondering: since the new project direction seems to be a whole lot more open to new ideas, maybe a new submenu system would be something to consider? @Owlet7 are you involve enough in that community to propose something? Would it even make` it better than workarounds?

**Comment by Owlet7** [comment]
I don't think that sub-episodes were a thing that Graf ever vetoed, but I would understand if he did. Implementing something like this in the engine would require rewriting a lot of stuff, which is work that probably wouldn't really be useful to the wider ZDoom community. It definitely would make it better for WadFusion, but I doubt anybody would wanna work on it.

There are some other features that Graf _did_ veto, which would be useful both for Fusion, and the wider ZDoom community, which I'm gonna suggest (or try implementing myself). Sub-episodes could also be suggested, but I doubt it'd go anywhere.

Just for the record, I don't believe that Graf was some kind of evil presence that loomed over ZDoom and prevented anything nice from happening. He's not exactly a nice guy and his leadership was terrible, but he gets vilified way more than he deserves.

**Comment by Trimint123** [comment]
> Just for the record, I don't believe that Graf was some kind of evil presence that loomed over ZDoom and prevented anything nice from happening. He's not exactly a nice guy and his leadership was terrible, but he gets vilified way more than he deserves.

I also don't see him as those kind of guy, just had very, _very_ questionable decision that leads to the fork of UZDoom. Regardless of what he did, he still responsible for what made GZDoom it is today, "credit where credit is due" as they say.

**Comment by maverick-hunter** [comment]
> > Just for the record, I don't believe that Graf was some kind of evil presence that loomed over ZDoom and prevented anything nice from happening. He's not exactly a nice guy and his leadership was terrible, but he gets vilified way more than he deserves.
> 
> I also don't see him as those kind of guy, just had very, _very_ questionable decision that leads to the fork of UZDoom. Regardless of what he did, he still responsible for what made GZDoom it is today, "credit where credit is due" as they say.

All I know is what people posted in the UZDoom announcement thread and that's pretty much what I got reading all the comments - the guy was responsible for invaluable progress, but was also kind of an a-hole to new people or ideas he didn't think would fit his view? I can't really tell anymore just from comments, but the whole "let's make UZDoom so people can discuss ideas instead of just shutting them down at birth" seemed interesting. 

Regarding the sub-episodes: wouldn't these also be useful for mods? Probably a niche thing though, right?

**Comment by Ohthatguy43** [comment]
> Here's a new test version, which should be near final. I fully expect it to break many mods, but if testing shows that it's not as bad as I imagine, I may integrate it with the main branch.
> 
> The latest dev builds of WadFusion and GZDoom are required.
> 
> There is one major bug, though, which makes me hesitate about integrating it—if you open the original New Game menu (e.g. from Fusion's compatibility options menu), don't start a new game from it, then try to start a new game from the custom New Game menu, GZDoom will crash. If you do start a new game from the original menu, it'll break the custom one, but the game won't crash.
> 
> [wadfusion_newgame_menu_test5.zip](https://github.com/user-attachments/files/20108315/wadfusion_newgame_menu_test5.zip)
> 
> "Hey, what happened to tests 3 and 4?" Don't worry about it :)

Sorry to bump this one, but will there be a new version of this?

**Comment by Owlet7** [comment]
I haven't had the opportunity and / or will to work on Fusion for a little while, but this is still a planned feature.

**Comment by Ohthatguy43** [comment]
> I haven't had the opportunity and / or will to work on Fusion for a little while, but this is still a planned feature.

I see, thanks for the quick reply.
